packages/meteor-wallet/src/lib/meteor-wallet.ts

```diff
@@ -0,0 +1,223 @@
+import {
+  connect,
+  keyStores,
+  transactions as nearTransactions,
+  utils,
+} from "near-api-js";
+import {
+  InjectedWallet,
+  Network,
+  Optional,
+  Transaction,
+  WalletBehaviourFactory,
+  WalletModuleFactory,
+} from "@near-wallet-selector/core";
+import {
+  EMeteorWalletSignInType,
+  IOMeteorWalletSdk_RequestSignIn_Inputs,
+  MeteorWallet as MeteorWalletSdk,
+} from "@meteorwallet/sdk";
+import { notNullEmptyArray } from "./utils/basic_utils";
+import {
+  MeteorWalletParams_Injected,
+  MeteorWalletState,
+} from "./meteor-wallet-types";
+import { createAction } from "@near-wallet-selector/wallet-utils";
+
+const setupWalletState = async (
+  params: MeteorWalletParams_Injected,
+  network: Network
+): Promise<MeteorWalletState> => {
+  const keyStore = new keyStores.BrowserLocalStorageKeyStore();
+
+  const near = await connect({
+    keyStore,
+    ...network,
+    headers: {},
+  });
+
+  const wallet = new MeteorWalletSdk({ near, appKeyPrefix: "near_app" });
+
+  // Cleanup up any pending keys (cancelled logins).
+  if (!wallet.isSignedIn()) {
+    await keyStore.clear();
+  }
+
+  return {
+    wallet,
+    keyStore,
+  };
+};
+
+const createMeteorWalletInjected: WalletBehaviourFactory<
+  InjectedWallet,
+  { params: MeteorWalletParams_Injected }
+> = async ({ options, logger, store, params }) => {
+  const _state = await setupWalletState(params, options.network);
+
+  const cleanup = () => {
+    _state.keyStore.clear();
+  };
+
+  const getAccounts = () => {
+    const accountId = _state.wallet.getAccountId();
+
+    if (!accountId) {
+      return [];
+    }
+
+    return [{ accountId }];
+  };
+
+  const transformTransactions = async (
```

Comment:
It looks like we're assuming `signerId` will be the account we're signed into. While this _should_ be case, ideally we'll want to validate this and throw to avoid potentially cryptic errors

Reply:
Yea probably for the best. In this case should I just loop through each transaction and compare the signer ID to the current account ID? What about in the case where `signerId` is undefined, is that even a possibility? (Maybe I should change the type in the case that its not)

Reply:
So the `signerId` should default to the active `accountId` (take a look at some of the other wallets such Ledger where we use `getActiveAccount`)

Reply:
I've added a check to the transactions and it seems to work.